pubsub/README.rst

```diff
@@ -113,9 +111,35 @@ Here's a few hints:
     same as ``ack_window`` but for nack requests
 
 
-``subscribe`` has also an optional ``metrics_client`` argument. You can provide
-any metrics agent that implements the same interface as ``MetricsAgent``
-(Datadog client will do ;) ) and get the following metrics:
+Prometheus Metrics
+^^^^^^^^^^^^^^^^^^
+
+If you like pull-based metrics like Prometheus you will be pleased to know that
+the subscriber records Prometheus metrics in the form
+``gcloud_aio_pubsub_<metric>``, which will have no effect if you don't use
+Prometheus to scrape app metrics:
+
+- ``subscriber_consume_total`` (labels: ``outcome = {'succeeded', 'cancelled',
+  'failed', 'failfast'}``) - [counter] a consume operation has completed with a
+  given outcome
+- ``subscriber_consume_latency_seconds`` (labels: ``phase = {'receive',
+  'queueing', 'runtime'}``) - [histogram] how many seconds taken to receive a
+  message, while waiting for processing, or to complete the callback
+- ``subscriber_batch_status_total`` (labels: ``component = {'acker', 'nacker'},
```

Comment:
Something about the naming of `subscriber_batch_status_total` is not sitting quite right with me. I think it is that, without looking at the labels, one would (or at least, I would) tend to assume this relates to batches pulled by the subscriber itself, when really it is only pertaining to messages being acked and nacked. I wonder, is there a word which encapsulates both `acker` and `nacker` which we could use to make this clearer? Perhaps even `subscriber_batch_ack_status_total` which still includes both `acker` and `nacker` as components. Not sure how I feel about that just yet. Of course, the answer might just be "just always look at the labels", but it would be great if this was less misleading.

Reply:
I'm not quite sure I agree, since the decision to `nack` or `ack` is also one that the user can decide when they write their callback. Either of the outcomes can be seen as "success" and can be considered "one more item that the PubSub client has processed" either successfully or unsuccessfully. Summing this metric should give the number of messages that were received in total (now that I've split out the "received" metric from the "processed" metric). Maybe the "processed" metric should die and this one should take its place instead? It is largely duplicating its efforts.

Reply:
Actually, my mental context was out-of-date and `processed` should continue to live on since it counts the actual number of messages passed through the acker or nacker and the `batch_status` counts the number of batches (of one or more messages).

Reply:
Sorry all I meant was that `BATCH_STATUS` and `BATCH_SIZE` are not necessarily referring to the same batch, which is fine, just that it could be misleading, hence the suggestion for the rename.